mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala

```diff
@@ -261,17 +290,40 @@ class CrossValidatorModel private[ml] (
     val copied = new CrossValidatorModel(
       uid,
       bestModel.copy(extra).asInstanceOf[Model[_]],
-      avgMetrics.clone())
+      avgMetrics.clone(),
+      CrossValidatorModel.copySubModels(subModels))
     copyValues(copied, extra).setParent(parent)
   }
 
   @Since("1.6.0")
   override def write: MLWriter = new CrossValidatorModel.CrossValidatorModelWriter(this)
+
+  @Since("2.3.0")
+  @throws[IOException]("If the input path already exists but overwrite is not enabled.")
+  def save(path: String, persistSubModels: Boolean): Unit = {
+    write.asInstanceOf[CrossValidatorModel.CrossValidatorModelWriter]
+      .persistSubModels(persistSubModels).save(path)
+  }
```

Comment:
I add this method because the `CrossValidatorModelWriter` is private. User cannot use it. But I don't know whether there is better solution.

Reply:
I tried `model.write.asInstanceOf[CrossValidatorModel.CrossValidatorModelWriter]` but cannot pass complier, it is inaccessible.
Do you have some other ways ?

Reply:
Discussion: Another way I think is adding an interface `def option(key: String, value: String)` into Writer. cc @jkbradley

Reply:
I agree with the last suggestion of adding ```def option(key: String, value: String)``` to mimic the SQL datasource API.